libdnf/utils/utils.cpp

```diff
@@ -166,7 +166,7 @@ bool haveFilesSameContent(const char * filePath1, const char * filePath2)
                 ret = false;
                 break;
             }
-        } while (readed > BLOCK_SIZE);
+        } while (readed == BLOCK_SIZE);
```

Comment:
Unless I'm missing something, this still doesn't compare the very end of the files if their size isn't a multiple of `BLOCK_SIZE`.

Somewhat unrelated to this patch, but I find complete lack of any error handling quite distressing. Seems this function needs much more fixing than just this?

Some nitpicks about the commit message: The way you phrase it it sounds like comparing the first "block" was actually intended functionality, while it in fact it was probably a serious bug. Also, when you say "block" there, it kind of implies filesystem block, but in reality its just how the buffer size is called here and is an unrelated arbitrary number...

Reply:
@lukash 
> Unless I'm missing something, this still doesn't compare the very end of the files if their size isn't a multiple of LOCK_SIZE.

Why do you think so? The last block of files is compared and then the loop ends because `readed != BLOCK_SIZE` (in case of their size is multiple of BLOCK_SIZE, `readed` is equal to 0).

Reply:
Right, I did miss that, sorry.

So just the lack of error handling remains...

Reply:
@lukash Do you want to handle an exception when fd1 = open(filePath1, 0)) fails? Or did you mean something else?
Additionally, the pull request could be handled conservatively. The new code did not create any new problem therefore we it can be merged.